sunpy/map/mapbase.py

```diff
@@ -2362,7 +2365,20 @@ def draw_contours(self, levels, axes=None, **contour_args):
             # Mask out the data array anywhere the coordinate arrays are not finite
             data = np.ma.array(data, mask=~np.logical_and(np.isfinite(x), np.isfinite(y)))
 
-        cs = axes.contour(x, y, data, levels, **contour_args)
+        if fill:
+            # Ensure we have more than one level if fill is True
+            if len(levels) == 1:
+                max_val = np.nanmax(self.data)
+                # Ensure the existing level is less than max_val
+                if levels[0] < max_val:
+                    levels = np.append(levels, max_val)
```

Comment:
I made the assumption the return from `_process_args` is always an array, I am not sure if that is true.

Reply:
The `_process_levels_arg` function accepts both integer and array-like inputs. When given an integer `n`, it computes up to `n+1` evenly spaced contour levels based on the minimum and maximum values in the data (`Z`). If an array-like input is provided, it directly uses these values as the contour levels. For `contourf`, specifically, providing an array-like input with only one element will lead to an error, as `contourf` requires at least two levels to define the areas to be filled between the contours.